lampo-cli/src/args.rs

```diff
@@ -1,163 +1,182 @@
+        amount_msat: Option<u64>,
+
+        #[arg(short, long)]
+        description: String,
+
+        #[arg(long, default_value = "3600")]
+        expiry: u32,
+    },
+
+    PayInvoice {
+        #[arg(short, long)]
+        invoice: String,
+        #[arg(long)]
+        amount_msat: Option<u64>,
+    },
+
+    Call {
+        method: String,
+        #[arg(allow_hyphen_values = true)]
+        params: Vec<String>,
+    },
+}
 
-#[derive(Debug)]
 pub struct LampoCliArgs {
     pub socket: String,
     pub method: String,
     pub args: HashMap<String, json::Value>,
 }
 
-struct Help {
-    name: &'static str,
-    description: &'static str,
-    version: &'static str,
-    usage: &'static str,
-}
+pub fn parse_args() -> error::Result<LampoCliArgs> {
+    let cli = Cli::parse();
+
+    // If data-dir is specified and socket is not specified,
+    // we need to get the socket path from it
+    // by appending the network name (default: testnet) to the path
+    // and adding the socket path (lampod.socket)
+    let socket = match cli.socket {
+        Some(s) => s,
+        None => {
+            let data_dir = cli.data_dir.unwrap_or_else(|| {
+                #[allow(deprecated)]
+                std::env::home_dir()
+                    .map(|path| path.to_string_lossy().to_string())
+                    .unwrap_or_else(|| ".".to_string())
+            });
+            let data_dir = format!("{data_dir}/.lampo");
+            format!("{}/{}/lampod.socket", data_dir, cli.network)
+        }
+    };
 
-const HELP: Help = Help {
-    name: "lampod-cli",
-    description: "Lampo Daemon command line",
-    version: env!("CARGO_PKG_VERSION"),
-    usage: r#"
-Usage
-
-    lampod-cli [<option> ...] <method> [arg=value]
-
-Options
-
-    -d | --data-dir     Specify lampo data directory (used to get socket path)
-    -n | --network      Set the network for lampo (default: testnet)
-    -s | --socket       Specify Unix Socket patch of the lampod node directely
-    -h | --help         Print help
-"#,
-};
-
-pub fn parse_args() -> Result<LampoCliArgs, lexopt::Error> {
-    use lexopt::prelude::*;
-
-    let mut data_dir: Option<String> = None;
-    let mut network: Option<String> = None;
-    let mut socket: Option<String> = None;
-    let mut method: Option<String> = None;
-    let mut args = HashMap::<String, json::Value>::new();
-
-    let mut parser = lexopt::Parser::from_env();
-    while let Some(arg) = parser.next()? {
-        match arg {
-            Short('d') | Long("data-dir") => {
-                let val: String = parser.value()?.parse()?;
-                data_dir = Some(val);
+    let (method, args) = match cli.command {
+        Commands::GetInfo => ("getinfo".to_string(), HashMap::new()),
+
+        Commands::ListChannels => ("listchannels".to_string(), HashMap::new()),
+
+        Commands::OpenChannel {
+            node_id,
+            capacity,
+            push_msat,
+            announce,
+        } => {
```

Comment:
we can remove the spaces?
```suggestion
    let (method, args) = match cli.command {
        Commands::GetInfo => ("getinfo".to_string(), HashMap::new()),
        Commands::ListChannels => ("listchannels".to_string(), HashMap::new()),
        Commands::OpenChannel {
            node_id,
            capacity,
            push_msat,
            announce,
        } => {
```